app/order-history/page.tsx

```diff
@@ -0,0 +1,381 @@
+          >
+            {paymentStatusFilter}
+            <button
+              onClick={() => setPaymentStatusFilter(null)}
+              className="ml-2 text-gray-500 hover:text-gray-700"
+            >
+              <X className="h-4 w-4" />
+            </button>
+          </div>
+        )}
+
+        {/* Clear all filters Button */}
+        <button
+          onClick={clearFilters}
+          disabled={!filter && serviceFilter.length === 0 && !logisticsFilter && !statusFilter && !paymentStatusFilter}
+          className={`ml-2 px-4 py-2 rounded-lg transition-all ${filter || serviceFilter.length > 0 || logisticsFilter || statusFilter || paymentStatusFilter ? "bg-[#9CC0D9] text-white hover:bg-blue-700" : "bg-gray-300 text-gray-500 cursor-not-allowed"}`}
+        >
+          Clear All
+        </button>
+      </div>
+
+      <div className="border border-gray-300 shadow-lg rounded-2xl overflow-hidden mt-6 p-4">
+
+      {/* Table of Orders */}
+      <Table className="min-w-full table-auto rounded-2xl overflow-hidden">
+
+        {/* Table Headers */}
+        <thead
+          className="text-left text-gray-700 uppercase tracking-wide text-sm font-semibold rounded-t-2xl"
+          style={{ backgroundColor: '#DBEAFF' }}
+        >
+          <tr className="border-b border-gray-200">
+            {tableHeaders.map((header, index) => (
+              <th key={index} className="py-2 px-4">{header}</th>
+            ))}
+          </tr>
+        </thead>
+
+        {/* Table Body */}
+        <tbody className="text-gray-600">
+          {filteredOrders.length > 0 ? (
+            filteredOrders.map((order, index) => ( //Map for filters
+              <tr key={index} className="border-b border-gray-200">
+
+                {/* Order Details */}
+                <td className="py-2 px-4">{order.orderId}</td>
+                <td className="py-2 px-4">{order.name}</td>
+                <td className="py-2 px-4">{order.address}</td>
+                <td className="py-2 px-4">{order.serviceType}</td>
+                <td className="py-2 px-4">{order.bag}</td>
+                <td className="py-2 px-4">{order.weight}</td>
+                <td className="py-2 px-4">{order.logistics}</td>
+                <td className="py-2 px-4">
+
+                  <Badge className={getBadgeColor(order.status)}>{order.status}</Badge>
+                </td>
+                <td className="py-2 px-4">
+                  <Badge className={getBadgeColor(order.paymentStatus)}>{order.paymentStatus}</Badge>
+                </td>
+                <td className="py-2 px-4">{order.date}</td>
+              </tr>
+            ))
+          ) : (
+            <tr>
+
+              {/* Case of 0 orders */}
+              <td colSpan={tableHeaders.length} className="text-center py-4">
+                No orders found.
+              </td>
+            </tr>
+          )}
+        </tbody>
```

Comment:
you used shadcn's table but not the `TableHeader` `TableBody` of shadcn, using them might have been easier para not that manual, pero we can go with this. just extract this to a component again, maybe something like `OrderHistoryTable.tsx`